packages/discord-bot/src/handlers/announce.ts

```diff
@@ -0,0 +1,100 @@
+import { UserAccountModel } from 'api/dist/useraccount/entities';
+import { UserAccount } from 'api/src/useraccount/types';
+import { UserModel } from 'api/dist/user/entities';
+import {
+  confirmButton,
+  cancelButton,
+} from '../utils/buttons/confirmationButtons';
+import { dmTargetMenu } from '../utils/menus/dmTargetmenu';
+import {
+  Message,
+  CommandInteraction,
+  SelectMenuInteraction,
+  MessageActionRow,
+} from 'discord.js';
+import { UserRole } from 'api/dist/user/types';
+import { dmTargets } from '../utils/dmTargets';
+
+export const announcementHandler = async (
+  interaction: CommandInteraction
+): Promise<void> => {
+  const { user } = interaction;
+  const ua = {
+    accountId: user.id,
+    name: user.username + '#' + user.discriminator,
+    avatarId: user.avatar,
+    platform: 'DISCORD',
+  } as UserAccount;
+  const userAccount = await UserAccountModel.findOneAndUpdate(
+    { accountId: user.id },
+    ua,
+    { upsert: true, new: true }
+  );
+  if (!userAccount.user) {
+    await interaction.editReply(
```

Comment:
Shouldn't you use `notActivatedError`, already defined, to ensure consistent error messages between the commands?